shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java

```diff
@@ -542,9 +543,12 @@ public void function(
                                 Function.identity(),
                                 name -> {
                                   CACHE_REFRESH.labels(readStale ? "cached" : "fresh", name).inc();
-                                  return inputSource
-                                      .fetch(name, readStale)
-                                      .collect(Collectors.toList());
+                                  final var results = inputSource.fetch(name, readStale);
+                                  if (results instanceof ErrorableStream<Object> errorableResults) {
+                                    if (!errorableResults.isOk())
+                                      throw new RuntimeException("Server error fetching data");
```

Comment:
Would it be possible to include the name of which input source errored? (In case there's a systematic error that needs to be tracked down)

Reply:
yes, good idea, fixed